docker/docker_image_dest.go

```diff
@@ -104,6 +122,19 @@ func (c *sizeCounter) Write(p []byte) (n int, err error) {
 	return len(p), nil
 }
 
+func prepareMountQuery(mounts []reference.Named, digest string) string {
+	var repositories []string
+	for _, m := range mounts {
+		// Mount only takes the path, not the hostname
+		mount := reference.Path(m)
```

Comment:
Still, shouldn’t this filter out any references where the registry hostname+port doesn’t match the current destination?

Reply:
… or maybe they should be refused, with an error, already in `SetMounts.

Anyway, if this moves into `SystemContext`, the location of the verification would be different.

Reply:
Ack, this changed a bit. LMK what you think.